src/hooks/usePools.ts

```diff
@@ -1,41 +1,111 @@
 import { useEffect, useState } from "react";
 import { getPerpetualProgramAndProvider } from "@/utils/constants";
 import { ProgramAccount } from "@project-serum/anchor";
+import { getTokenAddress, Token, tokenAddressToToken } from "@/lib/Token";
+import { PublicKey } from "@solana/web3.js";
+import { findProgramAddressSync } from "@project-serum/anchor/dist/cjs/utils/pubkey";
+import { Pool, TokenCustody } from "@/lib/Pool";
 
 export function usePools(wallet) {
-  const [pools, setPools] = useState<ProgramAccount<T>[]>([]);
-  const [custodies, setCustodies] = useState<Record<string, Object | Null[]>>(
-    {}
-  );
+  const [pools, setPools] = useState<Record<string, Pool>>();
+
+  let poolInfos = {};
 
   useEffect(() => {
     async function fetchPools() {
```

Comment:
mass fetching can be slow and frequent calls can hit rpc limits, please make sure you control frequency of the updates